### PR TITLE
Prometheus: Set interval on time field

### PR DESCRIPTION
**Pull request by toddtreece**
**What this PR does / why we need it**:

This adds the `Interval` field to the time field config for matrix responses in order to benefit from the changes added by https://github.com/grafana/grafana/pull/44622

Before:
image

After:
image

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
LGTM, but let someone who knows more about the Prometheus implications review also :)

**Review by gabor — COMMENTED** [review]
works fine in general, i have two minor issues below.

File: pkg/tsdb/prometheus/testdata/range_missing.result.golden.txt [hunk 1]
```diff
@@ -6,23 +6,17 @@ Frame[0] {
...
-| 2022-01-11 08:25:32 +0000 UTC | null                                           |
-| 2022-01-11 08:25:33 +0000 UTC | 21                                             |
-| 2022-01-11 08:25:34 +0000 UTC | 32                                             |
-| 2022-01-11 08:25:35 +0000 UTC | null                                           |
-| 2022-01-11 08:25:36 +0000 UTC | null                                           |
-| 2022-01-11 08:25:37 +0000 UTC | 43                                             |
-| 2022-01-11 08:25:38 +0000 UTC | null                                           |
+| 2022-01-11 03:25:33 -0500 EST | 21                                             |
+| 2022-01-11 03:25:34 -0500 EST | 32                                             |
+| 2022-01-11 03:25:37 -0500 EST | 43                                             |
```
[the remark this answers is not in the document]
Comment (toddtreece) [reply pkg/tsdb/prometheus/testdata/range_missing.result.golden.txt]: not sure what happened here either, but i deleted the file as suggested and it's in UTC now

File: pkg/tsdb/prometheus/time_series_query.go [hunk 2]
```diff
@@ -376,17 +317,18 @@ func matrixToDataFrames(matrix model.Matrix, query *PrometheusQuery, frames data
 		timeField := data.NewFieldFromFieldType(data.FieldTypeTime, len(v.Values))
 		valueField := data.NewFieldFromFieldType(data.FieldTypeNullableFloat64, len(v.Values))
 
-		for i, k := range v.Values {
-			timeField.Set(i, time.Unix(k.Timestamp.Unix(), 0).UTC())
-			value := float64(k.Value)
+		for i := 0; i < len(v.Values); i++ {
```
[the remark this answers is not in the document]
Comment (toddtreece) [reply pkg/tsdb/prometheus/time_series_query.go]: removed in [882c438](https://github.com/grafana/grafana/pull/44802/commits/882c4381efbe9f427fdd88bf9004f714d2ddc98e)

**Review by gabor — APPROVED** [review]
LGTM!

File: pkg/tsdb/prometheus/time_series_query_test.go [hunk 3]
```diff
@@ -631,16 +631,16 @@ func TestPrometheus_parseTimeSeriesResponse(t *testing.T) {
 			End:          time.Unix(4, 0).UTC(),
 			UtcOffsetSec: 0,
 		}
-		res, err := parseTimeSeriesResponse(value, query, true)
+		res, err := parseTimeSeriesResponse(value, query)
```
Comment (ivanahuckova) [inline pkg/tsdb/prometheus/time_series_query_test.go:634]: Do we still need this test as we are not null-filling anymore and the purpose of this test was to find out if nulls were inserted correctly?

**Review by ivanahuckova — COMMENTED** [review]
Left 1 comment, but great work with this! So much better!

File: pkg/tsdb/prometheus/time_series_query_test.go (hunk 3 above)
Comment (toddtreece) [reply pkg/tsdb/prometheus/time_series_query_test.go:634]: @ivanahuckova i switched the test to check that nulls are not inserted when there are gaps in the data. think that's worth keeping, or should i remove it?

**Comment by toddtreece** [comment]
> but great work with this

@ryantxu and @leeoniya deserve the credit for this one. thanks for the help!

**Comment by leeoniya** [comment]
now, let's not go around assignig ~blame~ credit to anyone 🤣

**Pull request merged**